src/psyclone/dynamo0p3.py

```diff
@@ -4515,11 +4511,48 @@ def __init__(self, fine_arg, coarse_arg):
         self.cell_map = sym.name
 
         # We have no colourmap information when first created
-        self.colourmap = ""
-        # Name of the variable holding the number of colours
-        self.ncolours_var = ""
-        # Name of the variable holding the last cell of a particular colour
-        self.last_cell_var = ""
+        self._colourmap_symbol = ""
+        # Symbol for the variable holding the number of colours
+        self._ncolours_var_symbol = ""
+        # Symbol of the variable holding the last cell of a particular colour
+        self._last_cell_var_symbol = None
+
+    def set_colour_info(self, colour_map, ncolours, last_cell):
+        '''Sets the colour_map, number of colours, and
+        last cell of a particular colour.
+
+        :param colour_map: the colour map symbol.
+        :type: colour_map:py:class:`psyclone.psyir.symbols.Symbol`
+        :param ncolours: the number of colours.
+        :type: ncolours: :py:class:`psyclone.psyir.symbols.Symbol`
+        :param last_cell: the last cell of a particular colour.
+        :type last_cell: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        '''
+        self._colourmap_symbol = colour_map
+        self._ncolours_var_symbol = ncolours
+        self._last_cell_var_symbol = last_cell
+
+    @property
+    def colourmap_symbol(self):
```

Comment:
Please could you add new explicit tests for these new getters and `set_colour_info`.

Reply:
Oh yes, of course.